packages/main/src/FileUploader.ts

```diff
@@ -317,6 +360,33 @@ class FileUploader extends UI5Element implements IFormInputElement {
 		}, "");
 	}
 
+	/**
+	 * Checks whether all files are below `maxFileSize` (if set),
+	 * and fires a `fileSizeExceeded` event if any file exceeds it.
+	 *
+	 * @returns true if all files are below `maxFileSize`
+	 * @private
+	 */
+	_validateFileSize(files: FileList | null): boolean {
+		if (!this.maxFileSize || !files) {
```

Comment:
Ideally, this `if` should not be part of this function. The `files` part of the `if` is already checked before calling the function in the first place, so it can be removed, and then `this.maxFileSize` can stay here, if you like, or be moved to the outside along with the other check, before calling the function

Reply:
Removed the files check but kept the max files check since `_validateFiles` is called in more than one place, and doesn't make sense to check if it exists every time it's called.